### PR TITLE
Raise Reset event when collection is cleared

### LINKED ISSUE #2856: RealmCollections should raise Reset when cleared
`RealmCollectionBase` raise `CollectionChanged` event with `Remove` action when cleared. Even though correct, it will be more efficient to raise `Reset` in this case, and we'd align with `ObservableCollections` too

Related to https://github.com/realm/realm-dotnet/issues/2854

### PR DESCRIPTION
**Pull request by papafe**
Fixes #2856 

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)

### CODE REVIEW

File: Realm/Realm/Linq/ChangeSet.cs [hunk 1]
```diff
@@ -61,13 +63,21 @@ public class ChangeSet
         /// <value>An array of <see cref="Move"/> structs, indicating the source and the destination index of the moved row.</value>
         public Move[] Moves { get; }
 
-        internal ChangeSet(int[] insertedIndices, int[] modifiedIndices, int[] newModifiedIndices, int[] deletedIndices, Move[] moves)
+        /// <summary>
+        /// Gets a value indicating whether the collection has been cleared.
+        /// The collection is cleared after calling methods such as <see cref="IList.Clear"/>.
+        /// </summary>
+        /// <value><c>true</c> if the collection has been cleared; <c>false</c> otherwise.</value>
+        public bool Cleared { get; }
```
Comment (fealebenpae) [inline Realm/Realm/Linq/ChangeSet.cs:71]:
```suggestion
        public bool IsCleared { get; }
```
I suggest the `Is<Adjective>` form to better match the rest of .NET.

**Review by fealebenpae — APPROVED** [review]

**Review by LaPeste — APPROVED** [review]
It seems reasonable to me

**Review by fealebenpae — APPROVED** [review]

**Comment by nirinchev** [comment]
@papafe can we merge this?

**Comment by papafe** [comment]
@nirinchev Probably yes. I've merged master into this, and if the CI is green I'll merge

**Pull request merged**